packages/forms-web-app/src/controllers/examination/enter-comment.js

```diff
@@ -1,28 +1,89 @@
+const config = require('../../config');
 const {
 	routesConfig: {
 		examination: {
 			directory: examinationDirectory,
 			pages: {
 				enterComment,
+				selectDeadline: { sessionIdPrimary, sessionIdSecondary, sessionIdTertiary },
+				selectFile: { route: selectFileRoute },
 				evidenceOrComment: { route: evidenceOrCommentRoute }
 			}
 		}
 	}
 } = require('../../routes/config');
+const examinationSessionStorage = config?.sessionStorage?.examination;
 
-const setData = {
+const pageData = {
 	backLinkUrl: `${examinationDirectory}${evidenceOrCommentRoute}`,
 	id: enterComment.id,
 	pageTitle: enterComment.name,
-	title: enterComment.name
+	title: enterComment.name,
+	captionTitle: 'Deadline item:'
 };
 
-const getEnterComment = (req, res) => {
-	res.render(enterComment.view, setData);
+const populatePageData = (hintText, comment = '') => ({
+	...pageData,
+	hint: 'Any further information requested by ' + hintText,
+	comment
+});
+
+const getActiveDeadlineItem = (examinationSession) =>
+	examinationSession[sessionIdPrimary]?.[sessionIdTertiary]?.[
+		examinationSession[sessionIdPrimary]?.[sessionIdSecondary]
+	];
+
+const getEnterComment = async (req, res) => {
+	const { session = {} } = req;
+	const examinationSession = session?.[examinationSessionStorage.name];
+
+	if (!examinationSession || !examinationSession[sessionIdPrimary])
+		return res.status(404).render('error/not-found');
```

Comment:
Not sure it is a great idea be throwing errors there and if there is an error boundary middleware that would gracefully handle them afterwards in their specific case.

We can verify it and if so we can raise a ticket to change it but not do this change in this one as it would break the pattern through all controllers we made previously and this change is out of the scope of this ticket itself...

Reply:
Would also be good to if we could find a solution to the jest mock for chainable functions this would be awesome actually.